lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -45,7 +45,7 @@ def authorize(money, payment, options = {})
         add_customer_data(request, options, payment)
         add_stored_credentials(request, options)
         post = { request: request, card: add_payment(payment, options) }
-        if options[:stored_credential]
+        if options[:stored_credential] || options[:payment_model]
```

Comment:
Can't be an `or` here, `:stored_credential` needs to be present, the point on the GSF is to provide the merchant with the posibility of overwrite the paymentModel, but the paymentReference still needs to be there.

Reply:
ok I see, I will remote it, thanks